x-pack/plugins/security_solution/public/management/pages/policy/view/policy_hooks.ts

```diff
@@ -79,7 +80,8 @@ export function usePolicyDetailsArtifactsNavigateCallback(listId: string) {
   );
 
   return useCallback(
-    (args: Partial<PolicyDetailsArtifactsPageLocation>) => history.push(getPath(args)),
+    (args: Partial<PolicyDetailsArtifactsPageLocation>, state?: PolicyDetailsRouteState) =>
```

Comment:
I might be missing something, but would it be worth having the `const { state: routeState } = useLocation();` in the `usePolicyDetailsArtifactsNavigateCallback` hook  instead of on each component? So when you use this hook, we get the current route state and you don't need an extra parameter here. What do you think?

Reply:
Interesting observation 👍  Isn't the callback called after the state changes? So we actually need the value saved prior to that, that is why it's being passed from the event context? I can't test it, but looking forward to seeing some answer from Konrad about it.

Reply:
Yeah, what I meant is having the `const { state: routeState } = useLocation();` in this hook, not inside the `useCallback`. So it can be used in the `useCallback` and don't need to pass it through params every time.  Sorry if I didn't explained well, but if I'm not wrong, we can avoid this new parameter and all the component level changes.

Reply:
Yeah that would definitely improve the code :) Thanks for explanation 👍

Reply:
@dasansol92 I went with this approach because I wasn't sure if in foreseeable future we won't need the hook to be used without passed state. I've made the change, now useLocation is being used directly in the hook.

Reply:
Yeah, now the change is limited to one file. Great suggestion @dasansol92, great implementation @szwarckonrad  :)